server_monitoring/models/server_monitor_process.py

```diff
@@ -0,0 +1,248 @@
+    def _default_uid(self):
+        return self.env.uid
+
+    @api.model
+    def _default_thread(self):
+        return threading.current_thread().name
+
+    @api.model
+    def _class_count(self):
+        counts = {}
+        context = self.env.context
+        if context.get('_x_no_class_count'):
+            return []
+        if context.get('_x_no_gc_collect'):
+            gc.collect()
+            gc.collect()
+        for obj in gc.get_objects():
+            if isinstance(obj, BLACKLIST):
+                continue
+            try:
+                cls = obj.__class__
+            except:
+                if isinstance(obj, types.ClassType):
+                    cls = types.ClassType
+                else:
+                    _logger.warning('Unknown object type for %r (%s)',
+                                    obj, type(obj))
+                    continue
+            name = '%s.%s' % (cls.__module__, cls.__name__)
+            try:
+                counts[name] += 1
+            except KeyError:
+                counts[name] = 1
+        info = []
+        for name, count in sorted(counts.items(),
+                                  key=itemgetter(1),
+                                  reverse=True):
+            if count < 2:
+                break
+            info.append({'name': name,  'count': count})
+        return [(0, 0, val) for val in info]
+
+    name = fields.Datetime('Timestamp',
+                           default=fields.Datetime.now,
+                           readonly=True)
+    pid = fields.Integer('Process ID',
+                         default=_default_pid,
+                         readonly=True,
+                         group_operator='count')
+    thread = fields.Char('Thread ID',
+                         default=_default_thread,
+                         readonly=True)
+    cpu_time = fields.Float(
+        'CPU time',
+        readonly=True,
+        default=_default_cpu_time,
+        group_operator='max',
+        help='CPU time consumed by the current server process')
+    memory = fields.Float(
+        'Memory',
+        default=_default_memory,
+        readonly=True,
+        group_operator='max',
+        help='Memory consumed by the current server process')
+    uid = fields.Many2one('res.users', 'User',
+                          default=_default_uid,
+                          readonly=True,
+                          select=True)
+    model = fields.Many2one('ir.model', 'Model',
+                            readonly=True,
+                            select=True)
+    method = fields.Char('Method', readonly=True)
+    status = fields.Char('RPC status', readonly=True)
+    sessionid = fields.Char('Session ID', readonly=True)
+    info = fields.Char('Information')
+    class_count_ids = fields.One2many(
+        'server.monitor.class.instance.count',
+        'measure_id',
+        'Class counts',
+        default=_class_count,
+        readonly=True)
+
+    _order = 'name DESC'
```

Comment:
Put this one at the beginning of class declaration, next to `_name`, `_description` and those kind of vars.